libkbfs/prefetcher.go

```diff
@@ -183,18 +185,55 @@ func (p *blockPrefetcher) applyToParentsRecursive(
 			panic(r)
 		}
 	}()
-	for b := range pre.parents {
-		parent, ok := p.prefetches[b]
+	for pptr := range pre.parents[ptr.RefNonce] {
+		parent, ok := p.prefetches[pptr.ID]
 		if !ok {
 			// Note that the parent (or some other ancestor) might be
 			// rescheduled for later and have been removed from
 			// `prefetches`.  In that case still delete it from the
 			// `parents` list as normal; the reschedule will add it
 			// back in later as needed.
-			delete(pre.parents, b)
+			delete(pre.parents[ptr.RefNonce], pptr)
```

Comment:
If `pre.parents[ptr.RefNonce]` becomes empty after this, do we bother deleting it entirely?

Reply:
Not for sure, but a future call to `applyToParentsRecursive` would delete it I think.  But maybe that's not guaranteed, so good point, I'll add that here.